api/src/index.js

```diff
@@ -19,18 +16,52 @@ const apiRouter = express.Router();
 
 app.use("/api", apiRouter);
 apiRouter.use("/nested", nestedRouter);
-apiRouter.use("/first-meal", firstMeal);
-apiRouter.use("/all-meals", allMeals);
-apiRouter.use("/future-meals", futureMeals);
-apiRouter.use("/future-meals", pastMeals);
-apiRouter.use("/last-meal", lastMeal);
 apiRouter.use("/meals", meals);
 apiRouter.use("/reservations", reservations);
+apiRouter.use("/reviews", reviews);
 
 app.get("/", async (req, res, next) => {
   res.send("Welcome");
 });
 
+apiRouter.get("/first-meal", async (req, res) => {
+  const firstMeal = await knex("Meal").orderBy("ID", "Asc").limit(1);
+  //Checking if the meal table is empty
+  if (firstMeal.length == 0) {
+    res.status(404).send("There is no registered meal");
+  } else {
+    res.status(200).json(firstMeal);
+  }
+});
+
+apiRouter.get("/last-meal", async (req, res) => {
+  const lastMeal = await knex("Meal").orderBy("ID", "desc").limit(1);
+  //Checking if the meal table is empty
+  if (lastMeal.length == 0) {
+    res.status(404).send("There is no registered meal");
+  } else {
+    res.status(200).json(lastMeal);
+  }
+});
+
+apiRouter.get("/future-meals", async (req, res) => {
+  const futureMeals = await knex("Meal").where("when", ">", knex.raw("NOW()"));
+  res.send(futureMeals);
+  res.end();
```

Comment:
I don't think you need to end explicitly. The send call should take care of that.

Reply:
Also it is better wrap the entire body of the function in a try-catch block, where you send error 500 in case anything goes wrong. That way you make sure that the client will get the correct response in case of an unexpected error.

Reply:
yes thats make sense no need to do it 2 times :D Thanks for pointing out